src/Microsoft.AspNet.Http.Interfaces/IRequestIdentifierFeature.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Http.Interfaces
+{
+    /// <summary>
+    /// Feature to identify a request.
+    /// </summary>
+    public interface IRequestIdentifierFeature
+    {
+        /// <summary>
+        /// Identifier to trace a request.
+        /// </summary>
+        Guid TraceIdentifier { get; }
```

Comment:
I'll concede that GUIDs are right from a *practical* perspective; but academically they are wrong from a *logical* perspective. (There's no absolute rule that says GUIDs are supposed to be ordered.)